readthedocs/doc_builder/director.py

```diff
@@ -615,6 +615,12 @@ def get_build_env_vars(self):
                 }
             )
 
+        env.update(
+            {
+                "READTHEDOCS_CANONICAL_URL": self.data.version.canonical_url,
```

Comment:
Can this be empty? Just wondering if we have some call to `log.error` anywhere in our code in case somehow we haven't defined this variable.

Reply:
No. This is a calculated value (uses `Project.get_docs_url` behind the scenes) and won't be empty.